### PR TITLE
Refactoring in mta package

### PR DESCRIPTION
**Pull request by radito3**
+ Add IntelliJ files to gitignore.

### CODE REVIEW

File: com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/PropertiesResolver.java [hunk 1]
```diff
@@ -197,7 +192,7 @@ private Object resolveKeyInIterable(String key, Collection<?> listOfProperties,
         if (StringUtils.isNumeric(key)) {
             try {
                 return IterableUtils.get(listOfProperties, Integer.parseInt(key));
-            } catch (IndexOutOfBoundsException e) {
+            } catch (IndexOutOfBoundsException ignored) {
```
Comment (petko-boyadzhiev) [inline com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/PropertiesResolver.java]: Please retain the E as a standard exception name and add a comment in the catch block explaining why do you ignore the exception

File: com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/ReferencePattern.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
     private String pattern;
     private String patternFormat;
 
-    private ReferencePattern(String pattern, String patternFormat) {
+    ReferencePattern(String pattern, String patternFormat) {
```
Comment (petko-boyadzhiev) [inline com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/ReferencePattern.java]: Why not private?

Comment (radito3) [reply com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/ReferencePattern.java]: By default, an enum's constructor is private. There is no need to explicitly put "private" here.

File: com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/PropertiesResolver.java (hunk 1 above)
Comment (radito3) [reply com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/PropertiesResolver.java]: I don't know. It was just ignored before :D

File: com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/model/VersionRule.java [hunk 3]
```diff
@@ -40,4 +25,16 @@ public boolean allows(DeploymentType deploymentType) {
     public static VersionRule value(String caseInsensitiveValue) {
         return VersionRule.valueOf(caseInsensitiveValue.toUpperCase());
     }
+
+    private static boolean isSameOrHigher(DeploymentType deploymentType) {
```
Comment (nictas) [inline com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/model/VersionRule.java]: The names of these methods are not OK. `isSameOrHigher` implies that there's some sort of >= comparison in the method when there isn't. `isUpgradable` is inconsistent with `isSameOrHigher` in terms of wording and `isAll` is not even grammatically correct.

File: com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/handlers/v3/ModulesSorter.java [hunk 4]
```diff
@@ -53,13 +52,13 @@ protected ModuleDependenciesCollector getDependenciesCollector() {
...
         List<Module> modules = new ArrayList<>(getModules());
-        Collections.sort(modules, getModuleComparator());
+        modules.sort(getModuleComparator());
         return modules;
     }
 
     protected List<Module> getModules() {
         List<Module> modules = new ArrayList<>(descriptor.getModules());
-        modules.stream()
+        modules
```
Comment (nictas) [inline com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/handlers/v3/ModulesSorter.java]: Format the code.

File: com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/PropertiesResolver.java (hunk 1 above)
Comment (petko-boyadzhiev) [reply com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/PropertiesResolver.java]: Then return the previous name, and write //Intentionally ignored

**Pull request merged**